### PR TITLE
chore: moving 'molgenis' wrapper to molgenis-components

### PR DESCRIPTION
**Pull request by mswertz**
todo: cannot get nuxt-ssr to run, help wanted

### CODE REVIEW

File: apps/molgenis-components/src/components/layout/MolgenisSignIn.vue [hunk 1]
```diff
@@ -26,131 +22,104 @@
...
   },
   methods: {
-    async signin() {
-      if (!this.email || !this.password) {
+    signin() {
+      if (this.email == null || this.password == null) {
         this.error = "Email and password should be filled in";
       } else {
         this.error = null;
-
```
Comment (mswertz) [inline apps/molgenis-components/src/components/layout/MolgenisSignIn.vue:79]: how does this compare to the 'request' client? would we need to use axios because of ssr?

**Review by mswertz — COMMENTED** [review]
nuxt still needs testing!

File: apps/molgenis-components/src/components/layout/MolgenisSignIn.vue (hunk 1 above)
Comment (connoratrug) [reply apps/molgenis-components/src/components/layout/MolgenisSignIn.vue:79]: yes, why would you change this ? this PR is not moving things forward

**Review by connoratrug — CHANGES_REQUESTED** [review]
remove request lib and use client of all requests

File: apps/molgenis-components/src/components/layout/MolgenisSignin.vue [hunk 2]
```diff
@@ -76,44 +71,32 @@ export default {
...
+        this.loading = true;
+        request(
+          "/api/graphql",
+          `mutation{signin(email: "${this.email}", password: "${this.password}"){status,message}}`
+        )
+          .then((data) => {
+            if (data.signin.status === "SUCCESS") {
+              this.success = "Signed in with " + this.email;
+              this.$emit("signin", this.email);
+              location.reload();
```
Comment (mswertz) [inline apps/molgenis-components/src/components/layout/MolgenisSignin.vue:83]: remove

File: apps/molgenis-components/public/assets/css/bootstrap-molgenis-blue.css [hunk 3]
```diff
@@ -0,0 +1,10196 @@
+.navbar {
```
[the remark this answers is not in the document]
Comment (mswertz) [reply apps/molgenis-components/public/assets/css/bootstrap-molgenis-blue.css:1]: not needed? than it can go indeed. In old version of styleguide we had this included for local use purposes

File: apps/molgenis-components/public/assets/css/bootstrap-molgenis-red.css [hunk 4]
```diff
@@ -0,0 +1,9328 @@
+.list-group-item, .table {
```
Comment (connoratrug) [inline apps/molgenis-components/public/assets/css/bootstrap-molgenis-red.css:1]: remove file

File: apps/molgenis-components/src/client/client.js [hunk 5]
```diff
@@ -1,4 +1,5 @@
 import axios from "axios";
+
```
Comment (connoratrug) [inline apps/molgenis-components/src/client/client.js:2]: remove to avoid extra file in diff

File: apps/molgenis-components/src/client/graphql.js [hunk 6]
```diff
@@ -0,0 +1,11 @@
+import axios from "axios";
+
+const request = async (url, graqphl) => {
+  const result = await axios.post(url, { query: graqphl }).catch((error) => {
+    console.log(JSON.stringify(error));
```
Comment (connoratrug) [inline apps/molgenis-components/src/client/graphql.js:5]: remove log

File: apps/molgenis-components/src/client/graphql.js [hunk 7]
```diff
@@ -0,0 +1,11 @@
+import axios from "axios";
```
Comment (connoratrug) [inline apps/molgenis-components/src/client/graphql.js:1]: no need to wrap client , this hides the api , and if you do want to wrap it , add it to the existing client instead of adding new file

File: apps/molgenis-components/src/components/layout/Molgenis.vue [hunk 8]
```diff
@@ -0,0 +1,210 @@
...
+            result[el] = url;
+          }
+        });
+      }
+      return result;
+    },
+    logo() {
+      if (this.logoURL) return this.logoURL;
+      else
+        return "/apps/molgenis-components/assets/img/molgenis_logo_white.png";
```
Comment (connoratrug) [inline apps/molgenis-components/src/components/layout/Molgenis.vue:147]: function should probably be called logoURL or logoURLString , and path should not be part of it , just keep de main img files an img folder

File: apps/molgenis-components/src/components/layout/Molgenis.vue [hunk 9]
```diff
@@ -0,0 +1,210 @@
...
+    },
+    css() {
+      if (this.cssURL) return this.cssURL;
+      else return "theme.css";
```
Comment (connoratrug) [inline apps/molgenis-components/src/components/layout/Molgenis.vue:151]: this.cssURL || "theme.css"

File: apps/molgenis-components/src/components/layout/MolgenisAccount.vue [hunk 10]
```diff
@@ -0,0 +1,116 @@
...
+        <InputPassword
+          id="account-password"
+          v-model="password"
+          label="Password"
+          placeholder="Enter password"
+          description="Please enter the password"
+          @enterPressed="updatePassword"
+        />
+        <InputPassword
+          id="account-password2"
```
Comment (connoratrug) [inline apps/molgenis-components/src/components/layout/MolgenisAccount.vue:18]: '-repeat' instead of '2'

File: apps/molgenis-components/src/components/layout/MolgenisSession.vue [hunk 11]
```diff
@@ -0,0 +1,220 @@
...
+      if (schemaSettings && schemaSettings._session) {
+        this.session = schemaSettings._session;
+      } else {
+        this.session = {};
+      }
+      //convert settings to object
+      this.session.settings = {};
+      if (dbSettings && dbSettings._settings) {
+        dbSettings._settings.forEach(
+          (s) =>
```
Comment (connoratrug) [inline apps/molgenis-components/src/components/layout/MolgenisSession.vue:137]: move to named function

File: apps/molgenis-components/src/components/layout/MolgenisSession.vue [hunk 12]
```diff
@@ -0,0 +1,220 @@
...
+                ? this.parseJson(s.value)
+                : s.value)
+        );
+        this.session.manifest = dbSettings._manifest;
+      }
+      // schemaSettings override dbSettings if set
+      if (schemaSettings && schemaSettings._settings) {
+        schemaSettings._settings.forEach(
+          (s) =>
+            (this.session.settings[s.key] =
```
Comment (connoratrug) [inline apps/molgenis-components/src/components/layout/MolgenisSession.vue:149]: code duplication , move to named ( and test)

File: apps/molgenis-components/src/components/layout/MolgenisSignin.vue [hunk 13]
```diff
@@ -131,8 +113,8 @@ export default {
 <docs>
 <template>
   <demo-item>
-    <SignInForm
-        :show="isShown"
```
Comment (connoratrug) [inline apps/molgenis-components/src/components/layout/MolgenisSignin.vue:135]: keep form in dom

File: apps/molgenis-components/src/client/graphql.js (hunk 7 above)
Comment (mswertz) [reply apps/molgenis-components/src/client/graphql.js:1]: I prefer to wrap client when using graphql. Indeed, this can be merged into client.js but for clarity I kept it seperate. No strong opinion so can merge

File: apps/molgenis-components/src/components/layout/MolgenisSignup.vue [hunk 14]
```diff
@@ -0,0 +1,149 @@
...
+        this.error = "Error: Passwords entered must be the same";
+      } else {
+        this.error = null;
+        this.loading = true;
+        request(
+          "/api/graphql",
+          `mutation{signup(email: "${this.email}", password: "${this.password}"){status,message}}`
+        )
+          .then((data) => {
+            console.log(JSON.stringify(data));
```
Comment (connoratrug) [inline apps/molgenis-components/src/components/layout/MolgenisSignup.vue:101]: remove debug code

File: apps/molgenis-components/vite.config.js [hunk 15]
```diff
@@ -10,7 +10,12 @@ let conf = {
   plugins: [docTagPlugin(), createVuePlugin()],
   server: {
     proxy: {
-      "^/.*/graphql": {
+      "^/graphql": {
```
Comment (connoratrug) [inline apps/molgenis-components/vite.config.js:13]: why is this needed ?

File: apps/nuxt-ssr/layouts/default.vue [hunk 16]
```diff
@@ -8,10 +8,10 @@
         <ButtonOutline v-else @click="showSignInForm = true" :light="true">
           Sign in
         </ButtonOutline>
-        <SignInForm
-          :show="showSignInForm"
+        <MolgenisSignin
+          v-if="showSignInForm"
```
Comment (connoratrug) [inline apps/nuxt-ssr/layouts/default.vue:12]: keep form in dom

File: apps/nuxt-ssr/store/index.js [hunk 17]
```diff
@@ -76,43 +76,33 @@ export const actions = {
...
-      .post('/api/graphql', {query})
-      .catch((error) => onSignInFailed('internal server graphqlError' + error));
-
-    if (signInResp.data.data.signin.status === 'SUCCESS') {
-      if (location && location.reload) {
-        location.reload();
-      }
-    } else {
-      onSignInFailed(signInResp.data.data.signin.message);
-    }
```
Comment (connoratrug) [inline apps/nuxt-ssr/store/index.js:102]: keep actions in actions

File: apps/molgenis-components/src/client/graphql.js (hunk 7 above)
Comment (mswertz) [reply apps/molgenis-components/src/client/graphql.js:1]: I expect in long run that we gonna move the whole logic of signin/signout to the client so this is just a temp situation. Meanwhile there are good use cases to want to have convenience function for graphql (instead of bothering every coder with the details.

File: apps/molgenis-components/src/client/graphql.js (hunk 6 above)
Comment (mswertz) [reply apps/molgenis-components/src/client/graphql.js:5]: done

**Review by connoratrug — CHANGES_REQUESTED** [review]
update pr title of is this still work in proggres ?

File: apps/molgenis-components/src/components/layout/Molgenis.vue [hunk 18]
```diff
@@ -0,0 +1,210 @@
...
+        //all databases
+        result["list all databases"] = "/";
+        this.session.schemas.forEach((s) => {
+          result[s] = "../../" + s; // all paths are of form /:schema/:app
+        });
+      }
+      return result;
+    },
+    crumbs() {
+      this.$route;
```
[the remark this answers is not in the document]
Comment (mswertz) [reply apps/molgenis-components/src/components/layout/Molgenis.vue:118]: please not now

File: apps/molgenis-components/src/components/layout/Molgenis.vue [hunk 19]
```diff
@@ -0,0 +1,210 @@
...
+      let path = decodeURI(
+        window.location.pathname.replace(location.search, "")
+      ).split("/");
+      let url = "/";
+      let result = {};
+      if (window.location.pathname != "/apps/central/") {
+        path.forEach((el) => {
+          if (el != "") {
+            url += el + "/";
```
[the remark this answers is not in the document]
Comment (mswertz) [reply apps/molgenis-components/src/components/layout/Molgenis.vue:127]: added checks. Also good subject to see how we want to do this in the future in a way that makes it very easy to configure and provides a 'convention over configuration' default.

File: apps/molgenis-components/public/assets/css/bootstrap-molgenis-blue.css (hunk 3 above)
[the remark this answers is not in the document]
Comment (mswertz) [reply apps/molgenis-components/public/assets/css/bootstrap-molgenis-blue.css]: done

File: apps/molgenis-components/public/assets/css/bootstrap-molgenis-red.css (hunk 4 above)
Comment (mswertz) [reply apps/molgenis-components/public/assets/css/bootstrap-molgenis-red.css]: done

File: apps/molgenis-components/src/components/layout/MolgenisAccount.vue (hunk 10 above)
Comment (mswertz) [reply apps/molgenis-components/src/components/layout/MolgenisAccount.vue:18]: done

File: apps/molgenis-components/src/components/layout/MolgenisAccount.vue [hunk 20]
```diff
@@ -0,0 +1,116 @@
...
+              this.success = "Success. Password changed";
+            } else {
+              this.error =
+                "Password change failed: " + data.changePassword.message;
+            }
+          })
+          .catch((error) => {
+            this.graphqlError = JSON.stringify(error);
+          });
+        this.loading = false;
```
[the remark this answers is not in the document]
Comment (mswertz) [reply apps/molgenis-components/src/components/layout/MolgenisAccount.vue:89]: ok

File: apps/molgenis-components/src/components/layout/MolgenisSession.vue [hunk 21]
```diff
@@ -0,0 +1,220 @@
...
+  },
+  props: {
+    graphql: {
+      default: "graphql",
+      type: String,
+    },
+  },
+  data: function () {
+    return {
+      /** @ignore */
```
[the remark this answers is not in the document]
Comment (mswertz) [reply apps/molgenis-components/src/components/layout/MolgenisSession.vue:83]: probably legacy for when we enabled sonar on the js. Removed.

File: apps/molgenis-components/src/components/layout/MolgenisSignin.vue (hunk 13 above)
Comment (connoratrug) [reply apps/molgenis-components/src/components/layout/MolgenisSignin.vue:135]: easer to debug and spot issues

File: apps/nuxt-ssr/layouts/default.vue (hunk 16 above)
Comment (connoratrug) [reply apps/nuxt-ssr/layouts/default.vue:12]: easer to debug and spot issues

**Comment by connoratrug** [comment]
chore ? suggest to select feat due to changes

**Comment by connoratrug** [comment]
running locally in dev mode the ssr app login work but the footer is messed up and a number of errors ar shown
Screenshot 2022-05-06 at 15 35 38

**Review by connoratrug — APPROVED** [review]
